packages/sdk38/src/secp256k1wallet.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { Secp256k1, Secp256k1Signature, Sha256 } from "@cosmjs/crypto";
+import { fromBase64, fromHex, toAscii } from "@cosmjs/encoding";
+
+import { extractKdfConfiguration, Secp256k1Wallet } from "./secp256k1wallet";
+import { base64Matcher, hexMatcher } from "./testutils.spec";
+import { executeKdf, KdfConfiguration } from "./wallet";
+
+describe("Secp256k1Wallet", () => {
+  // m/44'/118'/0'/0/0
+  // pubkey: 02baa4ef93f2ce84592a49b1d729c074eab640112522a7a89f7d03ebab21ded7b6
+  const defaultMnemonic = "special sign fit simple patrol salute grocery chicken wheat radar tonight ceiling";
+  const defaultPubkey = fromHex("02baa4ef93f2ce84592a49b1d729c074eab640112522a7a89f7d03ebab21ded7b6");
+  const defaultAddress = "cosmos1jhg0e7s6gn44tfc5k37kr04sznyhedtc9rzys5";
+
+  describe("fromMnemonic", () => {
+    it("works", async () => {
+      const wallet = await Secp256k1Wallet.fromMnemonic(defaultMnemonic);
+      expect(wallet).toBeTruthy();
+      expect(wallet.mnemonic).toEqual(defaultMnemonic);
+    });
+  });
+
+  describe("generate", () => {
+    it("defaults to 12 words", async () => {
+      const wallet = await Secp256k1Wallet.generate();
+      expect(wallet.mnemonic.split(" ").length).toEqual(12);
+    });
+
+    it("can use different mnemonic lengths", async () => {
+      expect((await Secp256k1Wallet.generate(12)).mnemonic.split(" ").length).toEqual(12);
+      expect((await Secp256k1Wallet.generate(15)).mnemonic.split(" ").length).toEqual(15);
+      expect((await Secp256k1Wallet.generate(18)).mnemonic.split(" ").length).toEqual(18);
+      expect((await Secp256k1Wallet.generate(21)).mnemonic.split(" ").length).toEqual(21);
+      expect((await Secp256k1Wallet.generate(24)).mnemonic.split(" ").length).toEqual(24);
+    });
+  });
+
+  describe("deserialize", () => {
```

Comment:
How about `serialize and deserialize`? Or have a test from a known serialization for `deserialize` and explicitly name this an integration test.

Reply:
Hmm, what I want to test here is "deserialize", so this should be highlighted. Having to execute serialize is a necessary evil. Maintaining a serialization in code can be done, but is a lot of work because of the encrypted data and comes from serialize() anyways.